src/ArmTemplates/Commands/Applications/RootCommandLineApplication.cs

```diff
@@ -0,0 +1,42 @@
+﻿// --------------------------------------------------------------------------
+//  <copyright file="RootCommandLineApplication.cs" company="Microsoft">
+//      Copyright (c) Microsoft Corporation. All rights reserved.
+//  </copyright>
+// --------------------------------------------------------------------------
+
+using System;
+using McMaster.Extensions.CommandLineUtils;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Commands.Abstractions;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.Azure.Management.ApiManagement.ArmTemplates.Commands.Applications
+{
+    /// <summary>
+    /// Built into DI registrar of command line commands.
+    /// </summary>
+    class RootCommandLineApplication : CommandLineApplication
+    {
+        readonly IServiceProvider serviceProvider;
+
+        public RootCommandLineApplication(IServiceProvider serviceProvider)
+        {
+            this.serviceProvider = serviceProvider;
+            this.RegisterCommandLineCommands();
+        }
```

Comment:
Instead of doing this, why not simply do this?
```suggestion
        public RootCommandLineApplication(List<ICommand> commands)
        : base(commands)
        {
        }
```

Reply:
Will resolve this due to deleting this code completely